src/v/security/jwt.h

```diff
@@ -114,4 +117,35 @@ class metadata {
     json::Document _metadata;
 };
 
+// A JSON Web Key Set as defined in
+// https://www.rfc-editor.org/rfc/rfc7517.html#section-5
+class jwks {
+public:
+    static result<jwks> make(std::string_view sv) {
+        json::Document doc;
+        doc.Parse(sv.data(), sv.length());
+        return make(std::move(doc));
+    }
+
+    static result<jwks> make(json::Document doc) {
+        if (doc.HasParseError() || !doc.IsObject()) {
+            return errc::jwks_invalid;
+        }
+        auto keys = doc.FindMember("keys");
+        if (keys == doc.MemberEnd() || !keys->value.IsArray()) {
+            return errc::jwks_invalid;
+        }
```

Comment:
question: Should you also ensure that the members names are unit per the RFC:

> The member names within a JWK Set MUST be unique; JWK Set parsers
   MUST either reject JWK Sets with duplicate member names or use a JSON
   parser that returns only the lexically last duplicate member name, as
   specified in [Section 15.12](https://datatracker.ietf.org/doc/html/rfc7517#section-15.12) ("The JSON Object") of [ECMAScript](https://datatracker.ietf.org/doc/html/rfc7517#ref-ECMAScript) 5.1
   [ECMAScript].

Reply:
Good spot, I'm changing the `emplace` to `insert_or_assign`.